### PR TITLE
Implement Oauth-2 and add test cases

### PR DESCRIPTION
**Pull request by harryi3t**

### CODE REVIEW

File: lib/authorizer/oauth2.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
+var HEADER = 'header',
+    URL = 'url',
+    DEFAULT_ADD_TOKEN_TO = HEADER,
```
Comment (kunagpal) [inline lib/authorizer/oauth2.js]: ?

File: lib/authorizer/oauth2.js [hunk 2]
```diff
@@ -44,6 +51,33 @@ module.exports = {
      * @param {AuthHandlerInterface~authSignHookCallback} done
      */
     sign: function (auth, request, done) {
+        var params = auth.get([
```
Comment (kunagpal) [inline lib/authorizer/oauth2.js]: Shift the array to a const above.

File: lib/authorizer/oauth2.js [hunk 3]
```diff
@@ -22,7 +29,7 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done
      */
     pre: function (auth, done) {
-        done(null, true);
+        done(null, Boolean(auth.get('accessToken')));
```
Comment (kunagpal) [inline lib/authorizer/oauth2.js]: Consider implementing `auth.has` to avoid repetitive `Boolean` casts.

File: test/unit/auth-handlers.test.js [hunk 4]
```diff
@@ -418,15 +418,98 @@ describe('Auth Handler:', function () {
...
+                    accessToken: '123456789abcdefghi',
+                    addTokenTo: 'header',
+                    tokenType: 'bearer'
+                }
+            },
+            url: 'https://api.github.com/user/orgs',
+            method: 'GET'
+        };
+
+        it('should sign the request by adding the token to the header', function () {
```
Comment (kunagpal) [inline test/unit/auth-handlers.test.js]: Also, add the corresponding logic and unit test for adding params to the request body.

File: lib/authorizer/oauth2.js (hunk 1 above)
Comment (harryi3t) [reply lib/authorizer/oauth2.js]: a token can be added to the header or to the URL
the default value will be `header`, if not specified on [line 64](https://github.com/postmanlabs/postman-runtime/pull/384/files#diff-7385a12d89075d66ee2143cd2e242ba5R64)

Comment (kunagpal) [reply lib/authorizer/oauth2.js]: So use `HEADER` there instead.

File: test/unit/auth-handlers.test.js (hunk 4 above)
Comment (harryi3t) [reply test/unit/auth-handlers.test.js]: params: see [line 448](https://github.com/postmanlabs/postman-runtime/pull/384/files#diff-ae634fae9ced1b620809db816fdab229R448)
body: not supported :(

File: lib/authorizer/oauth2.js (hunk 3 above)
Comment (harryi3t) [reply lib/authorizer/oauth2.js]: the check is not universal and can depend on the Auth and the parameter itself

File: test/unit/auth-handlers.test.js (hunk 4 above)
Comment (kunagpal) [reply test/unit/auth-handlers.test.js]: https://github.com/postmanlabs/postman-app-support/issues/3239

Comment (harryi3t) [reply test/unit/auth-handlers.test.js]: The said issue is about sending the `client_id` and `client_secret` in the header/body as part of 2nd request in Oauth2 to get the `access_token`
This PR is about using `access_token` for a protected resource

The rules for the token itself are stated here https://tools.ietf.org/html/rfc6750#section-2.1
Although bearer token can also be sent in the body, the RFC says that supporting it in the header is a MUST for server and have many conditions for a client to be able to add in the body.

File: lib/authorizer/oauth2.js [hunk 5]
```diff
@@ -1,3 +1,14 @@
+var HEADER = 'header',
+    URL = 'url',
```
Comment (kamalaknn) [inline lib/authorizer/oauth2.js]: Consider `param` or `urlParam`

**Pull request merged**